hardhat.tasks.js

```diff
@@ -117,42 +97,53 @@ task("register-controller", "registers controller with controller registry")
 
 task("ens-setApproval", "sets ens approval for podEnsRegistrar with ensHolder account")
   .addOptionalPositionalParam("controller")
-  .setAction(async (args, { getNamedAccounts, getChainId, ethers, deployments }) => {
+  .setAction(async (args, { getNamedAccounts, ethers, deployments, artifacts }) => {
     const { ensHolder } = await getNamedAccounts();
     const ensHolderSigner = ethers.provider.getSigner(ensHolder);
 
-    const network = await getChainId();
-    const ensRegistryAddress =
-      network === "31337" ? (await deployments.get("ENSRegistry")).address : getEnsAddress(network);
+    const ensRegistry = await ethers.getContractAtFromArtifact(
+      await artifacts.readArtifact("ENSRegistry"),
+      ensHolderSigner,
+    );
```

Comment:
❓ We're doing something similar on lines 151 and 311, should we remove the network checks in those cases too?

Reply:
oops good catch, we actually need those checks.